diff --git a/nixos/modules/services/monitoring/prometheus/exporters.nix b/nixos/modules/services/monitoring/prometheus/exporters.nix
index ef92e6469bca..02a49807beba 100644
--- a/nixos/modules/services/monitoring/prometheus/exporters.nix
+++ b/nixos/modules/services/monitoring/prometheus/exporters.nix
@@ -151,14 +151,6 @@ in
       options = (mkSubModules);
       imports = [
         ../../../misc/assertions.nix
-        (lib.mkRenamedOptionModule [ "unifi-poller" ] [ "unpoller" ])
-        (lib.mkRemovedOptionModule [ "minio" ] ''
-          The Minio exporter has been removed, as it was broken and unmaintained.
-          See the 24.11 release notes for more information.
-        '')
-        (lib.mkRemovedOptionModule [ "tor" ] ''
-          The Tor exporter has been removed, as it was broken and unmaintained.
-        '')
       ];
     };
     description = "Prometheus exporter configuration";
