diff --git a/nixos/modules/services/monitoring/prometheus/exporters.nix b/nixos/modules/services/monitoring/prometheus/exporters.nix
index c387bf06c8f6..e127a3103fc1 100644
--- a/nixos/modules/services/monitoring/prometheus/exporters.nix
+++ b/nixos/modules/services/monitoring/prometheus/exporters.nix
@@ -392,14 +392,6 @@ in
       options = mkSubModules;
       imports = [
         ../../../misc/assertions.nix
-        (lib.mkRenamedOptionModule [ "unifi-poller" ] [ "unpoller" ])
-        (lib.mkRemovedOptionModule [ "minio" ] ''
-          The Minio exporter has been removed, as it was broken and unmaintained.
-          See the 24.11 release notes for more information.
-        '')
-        (lib.mkRemovedOptionModule [ "tor" ] ''
-          The Tor exporter has been removed, as it was broken and unmaintained.
-        '')
       ];
     };
     description = "Prometheus exporter configuration";
