From 2d98b522fdde10078e7250026556788ad607e241 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maciej=20Kr=C3=BCger?= <mkg20001@gmail.com>
Date: Sat, 6 Sep 2025 14:24:19 +0200
Subject: [PATCH 3/3] ansible_2_{16,17}: fix build

---
 pkgs/top-level/all-packages.nix | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index c1a6a6b1cf8d..e8b1f4ba0dd5 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6432,6 +6432,22 @@ with pkgs;
         inherit version;
         hash = "sha256-Ob6KeYaix9NgabDZciC8L2eDxl/qfG1+Di0A0ayK+Hc=";
       };
+
+      postPatch = ''
+        substituteInPlace lib/ansible/executor/task_executor.py \
+          --replace "[python," "["
+
+        patchShebangs --build packaging/cli-doc/build.py
+
+        SETUPTOOLS_PATTERN='"setuptools[0-9 <>=.,]+"'
+        PYPROJECT=$(cat pyproject.toml)
+        if [[ "$PYPROJECT" =~ $SETUPTOOLS_PATTERN ]]; then
+          echo "setuptools replace: ''${BASH_REMATCH[0]}"
+          echo "''${PYPROJECT//''${BASH_REMATCH[0]}/'"setuptools"'}" > pyproject.toml
+        else
+          exit 2
+        fi
+      '';
     })
   );
   ansible_2_16 = python3Packages.toPythonApplication (
@@ -6441,6 +6457,22 @@ with pkgs;
         inherit version;
         hash = "sha256-gCef/9mGhrrfqjLh7HhdmKbfGy/B5Al97AWXZA10ZBU=";
       };
+
+      postPatch = ''
+        substituteInPlace lib/ansible/executor/task_executor.py \
+          --replace "[python," "["
+
+        patchShebangs --build packaging/cli-doc/build.py
+
+        SETUPTOOLS_PATTERN='"setuptools[0-9 <>=.,]+"'
+        PYPROJECT=$(cat pyproject.toml)
+        if [[ "$PYPROJECT" =~ $SETUPTOOLS_PATTERN ]]; then
+          echo "setuptools replace: ''${BASH_REMATCH[0]}"
+          echo "''${PYPROJECT//''${BASH_REMATCH[0]}/'"setuptools"'}" > pyproject.toml
+        else
+          exit 2
+        fi
+      '';
     })
   );
 
-- 
2.50.1

